### PR TITLE
Migrate CircleCI to Buildkite

### PR DESCRIPTION
**Pull request by sevanspowell**
# Description

- Use Nixkite to define a Buildkite pipeline in Nix.
- Modify the "test-ets.sh" script to reflect the new Buildkite
  environment.
- Remove old CircleCI config.
- Update the solidity compiler to version 0.5.1 because that's the
  version currently available in the Nix environment.

### CODE REVIEW

**Comment by sevanspowell** [comment]
Note that the CircleCI build check will fail because I've removed all the CircleCI config. I'll disable that check when everyone is happy with this PR.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# steps:
```
Comment (mirkoAlic) [inline .buildkite/pipeline.yml:1]: comments can be remove now? or the idea is to keep them for some reason? (minor)

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,100 +0,0 @@
...
-      - image: rtkaczyk/mantis-circleci:v6
-    steps:
-      - checkout
-
-      - restore_cache:
-          key: mantis-{{ checksum "build.sbt" }}-{{ checksum "project/build.properties" }}-{{ checksum "project/plugins.sbt" }}-v1
-
-      - run:
-          name: FULL ETS
-          command: RUN_FULL_ETS=yes ./test-ets.sh
```
Comment (mirkoAlic) [inline .circleci/config.yml:78]: Not for the scope of this task, but we should think how we could add this into our CI flow. Current main issue is that it take to much time.

**Review by mirkoAlic — APPROVED** [review]
LGTM!

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (sevanspowell) [reply .buildkite/pipeline.yml:1]: Sorry @mirkoAlic, they should be removed, I'll do that now :)

**Pull request merged**